src/test/compile-fail/match-defbm-explicit-mut.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
```

Comment:
So I've been advocating for us to organize the tests for new language features like this by putting them into a directory. So can we move this to ths directory:

```
src/test/compile-fail/rfc-XXX-default-binding-mode/
```

The idea is that we should be able to browse this directory (as well as corresponding directories in `src/test/run-pass` and `src/test/ui`) to get a good idea of how the feature behaves.